status-loop/queryValidators.js

```diff
@@ -72,25 +76,77 @@ function getStatusOfCampaign (campaign) {
+	const followerHbFromLeader = getRequest(`${validators[0].url}/channel/${campaign.id}/validator-messages/${validators[1].id}/Heartbeat?limit=15`)
+	const followerHbFromFollower = getRequest(`${validators[1].url}/channel/${campaign.id}/validator-messages/${validators[1].id}/Heartbeat?limit=15`)
 	const lastApproved = getRequest(`${validators[0].url}/channel/${campaign.id}/last-approved`)
 	const treePromise = getRequest(`${validators[0].url}/channel/${campaign.id}/validator-messages/${validators[0].id}/Accounting`)
 
-	return Promise.all([leaderHeartbeat, followerHeartbeat, lastApproved, treePromise])
-		.then(([leaderHbResp, followerHbResp, lastApprovedResp, treeResp]) => {
+	return Promise.all([leaderHb, followerHb, followerHbFromLeader, followerHbFromFollower, lastApproved, treePromise])
+		.then(([leaderHbResp, followerHbResp, followerHbFromLeaderResp, followerHbFromFollowerResp, lastApprovedResp, treeResp]) => {
 			const lastApproved = lastApprovedResp.lastApproved
 			const messagesFromAll = {
-				leaderHeartbeat: leaderHbResp.validatorMessages.map(x => x.msg).filter(x => x.type === 'Heartbeat'),
-				followerHeartbeat: followerHbResp.validatorMessages.map(x => x.msg).filter(x => x.type === 'Heartbeat'),
+				leaderHeartbeat: leaderHbResp.validatorMessages,
+				followerHeartbeat: followerHbResp.validatorMessages,
+				followerFromLeader: followerHbFromLeaderResp.validatorMessages,
+				followerFromFollower: followerHbFromFollowerResp.validatorMessages,
 				newStateLeader: lastApproved ? [lastApproved.newState.msg] : [],
 				approveStateFollower: lastApproved ? [lastApproved.approveState.msg] : []
 			}
 			const balanceTree = treeResp.validatorMessages[0] ? treeResp.validatorMessages[0].msg.balances : {}
-			return getStatus(messagesFromAll, campaign, balanceTree)
+			return {
+				status: getStatus(messagesFromAll, campaign, balanceTree),
+				lastHeartbeat: {
+					leader: getLasHeartbeatTimestamp(messagesFromAll.leaderHeartbeat[0]),
+					follower: getLasHeartbeatTimestamp(messagesFromAll.followerFromFollower[0])
+				}
+			}
 		})
 }
 
+function getLasHeartbeatTimestamp (msg) {
+	if (msg && msg.msg) {
+		return msg.msg.timestamp
+	} else {
+		return null
+	}
+}
+
+async function getDistributedFunds (campaign) {
+	const validators = campaign.spec.validators
+
+	const tree = await getRequest(`${validators[0].url}/channel/${campaign.id}/validator-messages/${validators[0].id}/Accounting`)
+	const balanceTree = tree.validatorMessages[0] ? tree.validatorMessages[0].msg.balances : {}
+	const totalBalances = Object.values(balanceTree).reduce((total, val) => total.add(new BN(val)), new BN(0))
+	const depositAmount = new BN(campaign.depositAmount)
+	const distributedFundsRatio = totalBalances.muln(1000).div(depositAmount) // in promiles
+
+	return +distributedFundsRatio.toString(10)
+}
+
+async function getEstimateInUsd (campaign) {
+	const factoryAddress = process.env.FACTORY_ADDR || null // will call default
```

Comment:
remove those env vars altogether, just call with null
we won't need to touch this (unless we deploy a custom uniswap, which may happen, but remove for now)